src/stories/containers/transparency-report/transparency-report.tsx

```diff
@@ -56,24 +57,64 @@ export const TransparencyReport = ({
 
   useEffect(() => {
     if (anchor) {
+      if (anchor.startsWith('forecast-')) {
+        setThirdIndex(TRANSPARENCY_IDS.indexOf('forecast'));
+        return;
+      }
       const index = TRANSPARENCY_IDS.indexOf(anchor);
       if (index > 0) {
         setThirdIndex(index);
       }
     }
   }, [anchor]);
 
+  const [scrolled, setScrolled] = useState<boolean>(false);
   useEffect(() => {
-    if (anchor && TRANSPARENCY_IDS.includes(anchor)) {
+    if (!scrolled && anchor && TRANSPARENCY_IDS.includes(anchor)) {
+      setScrolled(true);
       let offset = (transparencyTableRef?.current?.offsetTop || 0) - 280;
       const windowsWidth = Math.max(document.documentElement.clientWidth || 0, window.innerWidth || 0);
       if (windowsWidth < 834) {
         offset += 100;
       }
+      if ('scrollRestoration' in window.history) {
+        window.history.scrollRestoration = 'manual';
+      }
       window.scrollTo(0, Math.max(0, offset));
     }
   }, [anchor]);
 
+  useEffect(() => {
+    if (viewMonthStr) {
+      const month = DateTime.fromFormat(viewMonthStr as string, 'LLLyyyy');
+      setCurrentMonth(month);
+    }
+  }, []);
+
+  const replaceViewMonthRoute = (viewMonth: string) => {
+    router.replace({
+      hash: anchor,
+      query: {
+        ...router.query,
+        viewMonth
+      },
+    }, undefined, {
+      shallow: true,
+    });
+  };
+
+  const handlePreviousMonth = () => {
+    const month = currentMonth.minus({ month: 1 });
+    replaceViewMonthRoute(month.toFormat('LLLyyyy'));
+    setCurrentMonth(month);
+  };
+
+  const handleNextMonth = () => {
```

Comment:
What do you think about use useCallback hook for this functions??

Reply:
It's a good idea, I changed it